### PR TITLE
Add lending_pool_add_bank2 ix

### PR DESCRIPTION
**Pull request by jkbpvsc**
marginfi_group module

### CODE REVIEW

**Comment by jkbpvsc** [comment]
@exdx feel free to work on this PR

**Review by exdx — COMMENTED** [review]
Still WIP but close. 
Outstanding: CLI changes

File: test-utils/src/marginfi_group.rs [hunk 1]
```diff
@@ -123,6 +123,72 @@ impl MarginfiGroupFixture {
...
+                bank_config: bank_config.into(),
+            }
+            .data(),
+        };
+
+        let tx = Transaction::new_signed_with_payer(
+            &[ix],
+            Some(&self.ctx.borrow().payer.pubkey().clone()),
+            &[&self.ctx.borrow().payer],
+            self.ctx.borrow().last_blockhash,
```
[the remark this answers is not in the document]
Comment (exdx) [reply test-utils/src/marginfi_group.rs:180]: We do need to add a new ix for the PDA though -- working on that currently

File: programs/marginfi/tests/marginfi_group.rs [hunk 2]
```diff
@@ -181,6 +177,32 @@ async fn marginfi_group_add_bank_success() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn marginfi_group_add_bank_with_seed_success() -> anyhow::Result<()> {
+    // Setup test executor with non-admin payer
+    let test_f = TestFixture::new(None).await;
+
+    let bank_asset_mint_fixture = MintFixture::new(test_f.context.clone(), None, None).await;
+    let bank_seed = 1200_u64;
```
[the remark this answers is not in the document]
Comment (jkbpvsc) [reply programs/marginfi/tests/marginfi_group.rs:186]: No need fo tests

File: programs/marginfi/src/instructions/marginfi_group/add_pool.rs [hunk 3]
```diff
@@ -165,3 +165,169 @@ pub struct LendingPoolAddBank<'info> {
...
 }
+
+/// A copy of lending_pool_add_bank but with an additional bank seed provided.
+/// This seed is used by the LendingPoolAddBankWithSeed.bank to generate a
+/// PDA account to sign for newly added bank transactions securely.
+/// The previous lending_pool_add_bank is preserved for backwards-compatibility.
+pub fn lending_pool_add_bank_with_seed(
+    ctx: Context<LendingPoolAddBankWithSeed>,
+    bank_config: BankConfig,
+    bank_seed: u64,
```
[the remark this answers is not in the document]
Comment (jkbpvsc) [reply programs/marginfi/src/instructions/marginfi_group/add_pool.rs]: No it doesn't, it just needs to be passed in the top level fn sig (in lib.rs)

**Comment by exdx** [comment]
The commit history is a bit of a mess due to the gpg key setup not working correctly on my new x86 dev box. Will fix.

File: programs/marginfi/src/instructions/marginfi_group/add_pool.rs [hunk 4]
```diff
@@ -165,3 +165,169 @@ pub struct LendingPoolAddBank<'info> {
...
+        seeds = [
+            marginfi_group.key().as_ref(),
+            bank_mint.key().as_ref(),
+            &bank_seed.to_le_bytes(),
+        ],
+        bump,
+    )]
+    pub bank: AccountLoader<'info, Bank>,
+
+    /// CHECK: ⋐ ͡⋄ ω ͡⋄ ⋑a
```
Comment (jkbpvsc) [inline programs/marginfi/src/instructions/marginfi_group/add_pool.rs:261]:
```suggestion
    /// CHECK: ⋐ ͡⋄ ω ͡⋄ ⋑
```

**Comment by jkbpvsc** [comment]
LGTM
Waiting for a stamp from @losman0s then we can merge and deploy.

**Review by losman0s — APPROVED** [review]

**Comment by losman0s** [comment]
lgtm

I guess using the mint in the PDA derivation is not technically needed, since we're anyway adding the differentiator (`seed`) to avoid losing the ability to create several banks per mint. However it's not problematic either, and we can choose "logical" seeds for the deriv that way.

**Pull request merged**